velero-plugins/build/restore.go

```diff
@@ -47,6 +48,39 @@ func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*v
 	p.Log.Info(fmt.Sprintf("Found new dockercfg secret: %v", secret))
 	build = createNewPushSecret(build, secret)
 
+	// Swap out stale imageRefs
+	// TEMP: Getting registry information dynamically because annotations do not exist
+	version, err := common.GetServerVersion()
+	if err != nil {
+		p.Log.Error("Error getting server version: ", err)
+		return nil, err
+	}
+	registry, err := common.GetRegistryInfo(version.Major, version.Minor)
+	if err != nil {
+		p.Log.Error("Error getting registry information: ", err)
+		return nil, err
+	}
+	name := build.Spec.Strategy.SourceStrategy.From.Name
+	shaSplit := strings.Split(name, "@")
+	if len(shaSplit) < 2 {
+		err = fmt.Errorf("unexpected image reference [%v]", name)
+		return nil, err
+	}
+	sha := shaSplit[1]
+	splitName := strings.Split(shaSplit[0], "/")
+	if len(splitName) < 2 {
+		err = fmt.Errorf("unexpected image reference [%v]", name)
+		return nil, err
+	}
+	namespacedName := splitName[len(splitName)-2:]
+	newName := fmt.Sprintf("%s/%s/%s@%s", registry, namespacedName[0], namespacedName[1], sha)
```

Comment:
Once you get the annotation lookups working, I wonder if you need to compare splitname[0] to the backup registry path. It should be identical to the old registry. I'm not sure what we'd do if it's not though, since I'm assuming builds must always be local. Perhaps we log a warning but still replace the name to match the restore cluster path.

Reply:
Yeah I think that's a good idea. We should definitely flag a warning if the two don't match